### PR TITLE
Improve code quality using lgtm alerts

### PR DESCRIPTION
**Pull request by frafra**
## Checklist

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [X] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [X] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [X] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [X] New unit tests have been added covering the changes, unless there are explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [X] This PR passes the QA checks: flake8 geonode
- [X] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

Related to: #5710

### CODE REVIEW

**Comment by frafra** [comment]
There are ~200 alterts left. It is not necessary to fix all of them, but nor is mandatory to have all of them fixed at once :-)

I would suggest to start a review, so I am sure we are on the same page with respect to old cruft that could be removed (I tried not to be too harsh and keep some old code as comment sometimes with FIXME etc.).

After the code review, a merge can be done. I would suggest to fix other alerts in other PR, as this is quite hefty already and it becomes hard to understand what issue are left to be fixed due to lgtm interface.

**Comment by afabiani** [comment]
@frafra I see some alerts warning for syntax error for Python 2.7
Is this behavior intentional?

**Comment by frafra** [comment]
> @frafra I see some alerts warning for syntax error for Python 2.7
> Is this behavior intentional?

Are you looking at lgtm? If so, that are syntax errors in the current master branch that I fixed in this PR.

File: geonode/client/hooksets.py [hunk 1]
```diff
@@ -73,9 +73,6 @@ def map_download_template(self, context=None):
 
     # Map Persisting
     def viewer_json(self, conf, context=None):
-        if not context:
```
Comment (afabiani) [inline geonode/client/hooksets.py]: Can we please change this to

`def viewer_json(self, conf, context={}):`

File: geonode/client/hooksets.py [hunk 2]
```diff
@@ -73,9 +73,6 @@ def map_download_template(self, context=None):
 
     # Map Persisting
     def viewer_json(self, conf, context=None):
-        if not context:
-            context = {}
-
         if isinstance(conf, string_types):
             conf = json.loads(conf)
         return conf
```
Comment (afabiani) [inline geonode/client/hooksets.py]: Not sure of the consequences of returning `None` instead of `{}`.

I would revert this change, maybe by writing it a bit better.

File: geonode/layers/views.py [hunk 3]
```diff
@@ -698,12 +695,6 @@ def sld_definition(style):
     if request.user.has_perm(
         'download_resourcebase',
             layer.get_self_resource()):
-        if layer.storeType == 'dataStore':
-            links = layer.link_set.download().filter(
-                name__in=settings.DOWNLOAD_FORMATS_VECTOR)
-        else:
-            links = layer.link_set.download().filter(
-                name__in=settings.DOWNLOAD_FORMATS_RASTER)
```
Comment (afabiani) [inline geonode/layers/views.py]: Why we removed `links` variable?

File: geonode/themes/models.py [hunk 4]
```diff
@@ -97,7 +97,6 @@ class GeoNodeThemeCustomization(models.Model):
     contact_city = models.CharField(max_length=255, null=True, blank=True)
     contact_street = models.CharField(max_length=255, null=True, blank=True)
     contact_postal_code = models.CharField(max_length=255, null=True, blank=True)
-    contact_city = models.CharField(max_length=255, null=True, blank=True)
```
Comment (afabiani) [inline geonode/themes/models.py]: Are we sure we are not using this variable?

**Comment by afabiani** [comment]
Tryin to upload a layer I'm getting the following exception:

```
File "/mnt/d/work/code/python/geonode/geonode-2.10.x/geonode/security/utils.py", line 491, in set_owner_permissions
    for perm in models.LAYER_ADMIN_PERMISSIONS:
NameError: name 'models' is not defined
```

**Comment by afabiani** [comment]
Filters by owners and categories do not work anymore

image

**Comment by afabiani** [comment]
Error on api counts

```
Traceback (most recent call last):
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/tastypie/resources.py", line 276, in wrapper
    return self._handle_500(request, e)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/tastypie/resources.py", line 326, in _handle_500
    return self.error_response(request, data, response_class=response_class)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/tastypie/resources.py", line 1299, in error_response
    serialized = self.serialize(request, errors, desired_format)
  File "/mnt/d/work/code/python/geonode/geonode-2.10.x/geonode/api/api.py", line 276, in serialize
    return super(TopicCategoryResource, self).serialize(request, data, format, options)
  File "/mnt/d/work/code/python/geonode/geonode-2.10.x/geonode/api/api.py", line 147, in serialize
    return super(TypeFilteredResource, self).serialize(request, data, format, options)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/tastypie/resources.py", line 405, in serialize
    return self._meta.serializer.serialize(data, format, options)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/tastypie/serializers.py", line 271, in serialize
    return method(bundle, options)
  File "/mnt/d/work/code/python/geonode/geonode-2.10.x/geonode/api/api.py", line 106, in to_json
    counts = self.get_resources_counts(options)
  File "/mnt/d/work/code/python/geonode/geonode-2.10.x/geonode/api/api.py", line 97, in get_resources_counts
    resources = resources.filter(polymorphic_ctype__model=options['type_filter'])
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/query.py", line 892, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/polymorphic/query.py", line 173, in _filter_or_exclude
    negate, *(list(q_objects) + additional_args), **kwargs
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/modeltranslation/manager.py", line 307, in _filter_or_exclude
    return super(MultilingualQuerySet, self)._filter_or_exclude(negate, *args, **kwargs)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/query.py", line 910, in _filter_or_exclude
    clone.query.add_q(Q(*args, **kwargs))
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/sql/query.py", line 1290, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/sql/query.py", line 1318, in _add_q
    split_subq=split_subq, simple_col=simple_col,
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/sql/query.py", line 1251, in build_filter
    condition = self.build_lookup(lookups, col, value)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/sql/query.py", line 1116, in build_lookup
    lookup = lookup_class(lhs, rhs)
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/lookups.py", line 20, in __init__
    self.rhs = self.get_prep_lookup()
  File "/home/alessio/Envs/geonode-3.0/lib/python3.5/site-packages/django/db/models/lookups.py", line 68, in get_prep_lookup
    return self.rhs._prepare(self.lhs.output_field)
TypeError: _prepare() takes 1 positional argument but 2 were given
```

File: geonode/client/hooksets.py (hunk 2 above)
Comment (frafra) [reply geonode/client/hooksets.py]: It does not change the behavior of the function. `context ` has no effect.

File: geonode/layers/views.py (hunk 3 above)
Comment (frafra) [reply geonode/layers/views.py]: Dead code, not used afterwards.

File: geonode/themes/models.py (hunk 4 above)
Comment (frafra) [reply geonode/themes/models.py]: It is duplicated, it is defined twice. There are multiple fields in the models that are defined multiple times.

**Comment by frafra** [comment]
> Tryin to upload a layer I'm getting the following exception:
> 
> ```
> File "/mnt/d/work/code/python/geonode/geonode-2.10.x/geonode/security/utils.py", line 491, in set_owner_permissions
>     for perm in models.LAYER_ADMIN_PERMISSIONS:
> NameError: name 'models' is not defined
> ```

My bad. I know where that comes from. I was removing import loops.

File: geonode/client/hooksets.py (hunk 1 above)
Comment (frafra) [reply geonode/client/hooksets.py]: Sure, it seems a better default.
